sdk/core/azure-core/azure/core/credentials.py

```diff
@@ -3,21 +3,56 @@
 # Licensed under the MIT License. See LICENSE.txt in the project root for
 # license information.
 # -------------------------------------------------------------------------
-from typing import Any, NamedTuple, Optional
+from typing import Any, NamedTuple, Optional, TypedDict
 from typing_extensions import Protocol, runtime_checkable
 
 
 class AccessToken(NamedTuple):
     """Represents an OAuth access token."""
 
     token: str
+    """The token string."""
     expires_on: int
-    refresh_on: Optional[int] = None
+    """The token's expiration time in Unix time."""
 
 
-AccessToken.token.__doc__ = """The token string."""
-AccessToken.expires_on.__doc__ = """The token's expiration time in Unix time."""
-AccessToken.refresh_on.__doc__ = """When the token should be refreshed in Unix time."""
+class AccessTokenInfo:
```

Comment:
Let's also add is_pop into the token info.

Reply:
Should be `token_type` with the default being "bearer", right? (Based on https://github.com/Azure/azure-sdk-for-net/pull/45134/files#diff-fa2c390af9cd1cb54117b35c3ff3d08713ed339adbc4cf0659690f9663558714R46)

I think we can hold off on adding that until we add the rest of the PoP related parameters.

Reply:
Then we need token_type here, right?